File: medium/Hash-Table/contains-duplicate/contains_duplicate.cpp
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> st;
        for (int num : nums) {
            if (st.find(num) != st.end()) {
                return true;
            }
            st.insert(num);
        }
        return false;
    }
};

// Test
#include <iostream>
int main() {
    Solution sol;
    vector<int> nums = {1,2,3,1};
    cout << sol.containsDuplicate(nums) << endl;
    return 0;
}

File: medium/Hash-Table/contains-duplicate/contains_duplicate.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        """
        Check if the array has duplicate elements.
        Approach: Hash Set
        Time: O(n)
        Space: O(n)
        """
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False

if __name__ == "__main__":
    sol = Solution()
    print(sol.containsDuplicate([1,2,3,1]))  # True
